### PR TITLE
Internals interceptor

### PR DESCRIPTION
**Pull request by bergundy**
## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: packages/workflow/src/dependencies.ts [hunk 1]
```diff
@@ -70,3 +70,14 @@ export enum ApplyMode {
...
 }
+
+/**
+ * Call information for async external dependencies
+ */
+export interface ExternalCall {
+  ifaceName: string;
+  fnName: string;
+  args: any[];
+  /** Optional in case applyMode is ASYNC_IGNORED */
```
Comment (Sushisource) [inline packages/workflow/src/dependencies.ts:81]: Maybe expand on what one has to do with the other?

File: packages/workflow/src/interceptors.ts [hunk 2]
```diff
@@ -137,19 +138,55 @@ export interface WorkflowOutboundCallsInterceptor {
...
+   *
+   * Returns a `boolean` indicating whether this job was processed or not
+   */
+  activate?(input: ActivateInput, next: Next<this, 'activate'>): Promise<boolean>;
+  /**
+   * Called after all `WFActivationJob`s have been proccesed for an activation.
+   *
+   * Can manipulate the commands generated by the Workflow
+   */
+  concludeActivation?(input: ConcludeActivationInput, next: Next<this, 'concludeActivation'>): ConcludeActivationOutput;
```
Comment (Sushisource) [inline packages/workflow/src/interceptors.ts:172]: Should these be "activated" and "concluded"?

**Review by Sushisource — APPROVED** [review]

File: packages/workflow/src/dependencies.ts (hunk 1 above)
Comment (bergundy) [reply packages/workflow/src/dependencies.ts:81]: There's an entire page in the docs for this concept.

File: packages/workflow/src/interceptors.ts (hunk 2 above)
Comment (bergundy) [reply packages/workflow/src/interceptors.ts:172]: These are the names of the methods in the Workflow runtime, I'd rather keep them.

**Pull request merged**